src/dmd/parse.d

```diff
@@ -4715,36 +4740,90 @@ final class Parser(AST) : Lexer
             }
 
         case TOK.in_:
+            auto loc = token.loc;
```

Comment:
Add a couple comments presenting the supported `in` syntaxes as per `invariant` above.